### DISCUSSION TITLE
Is there a way to retrieve results from Milvus based on the threshold for euclidean distance

### DISCUSSION THREAD

**Discussion by ZubeenK** [milvus-io/milvus#21837] [Q&A and General discussion]
Is there a way to retrieve results from Milvus based on the threshold for euclidean distance?

**Comment by xiaofan-luan** [comment]
> Is there a way to retrieve results from Milvus based on the threshold for euclidean distance?

That will be released soon on 2.3 (Estimated end of Feb)

**Reply by xiaofan-luan** [reply]
we call it range search

**Reply by ZubeenK** [reply]
@xiaofan-luan So, in that feature, we will be able to get results within certain threshold (Euclidean distance), is my understanding correct?

**Reply by yhmo** [reply]
> @xiaofan-luan So, in that feature, we will be able to get results within certain threshold (Euclidean distance), is my understanding correct?

Right, we will provide two parameters for search():
- A parameter to define the outer threshold
- An optional parameter to define the inner threshold
- Be careful, the returned results are limited, the max value is 16384.

**Reply by i14d4** [reply]
@xiaofan-luan  and @yhmo : Great to hear that you work on an option for a distance threshold - thanks a lot for the information ! 
Is there any update when the range search / (included in release 2.3?) will be released? Thanks a lot in advance !

**Reply by yhmo** [reply]
> @xiaofan-luan and @yhmo : Great to hear that you work on an option for a distance threshold - thanks a lot for the information ! Is there any update when the range search / (included in release 2.3?) will be released? Thanks a lot in advance !

Yes, the range search will be available on v2.3.0, which will be released in a couple of weeks.

**Reply by ZubeenK** [reply]
Any update on the range search feature?

**Reply by xiaofan-luan** [reply]
> Any update on the range search feature?

Will be released on 2.3. Already has a 2.3-RC1 but working on fixing stability issues. ETA end of may

**Reply by ZubeenK** [reply]
@xiaofan-luan Any update on the range search feature?

**Reply by xiaofan-luan** [reply]
This is already in the 2.3 release.
We will release 2.3 early July
